public/app/listeners/setDatasetFile.js

```diff
@@ -0,0 +1,37 @@
+const fs = require('fs');
+const logger = require('../logger');
+const { SET_DATASET_FILE_CHANNEL } = require('../../shared/channels');
+const {
+  SET_DATASET_FILE_SUCCESS,
+  SET_DATASET_FILE_ERROR,
+} = require('../../shared/types');
+const { ERROR_MISSING_FILE, ERROR_GENERAL } = require('../../shared/errors');
+const { DATASETS_COLLECTION } = require('../db');
+
+const setDatasetFile = (mainWindow, db) => async (e, { id, content }) => {
+  try {
+    const { filepath } = db.get(DATASETS_COLLECTION).find({ id }).value();
+
+    if (!fs.existsSync(filepath)) {
+      return mainWindow.webContents.send(
+        SET_DATASET_FILE_CHANNEL,
+        ERROR_MISSING_FILE,
+      );
+    }
```

Comment:
This is a better way of handling missing files than checking the `ENOENT` error code :+1: We should do that everywhere